### DISCUSSION TITLE
Microsoft.Extensions.ObjectPool

### DISCUSSION THREAD

**Discussion by Code-Grump** [reqnroll/Reqnroll#210] [General]
A lot of the discussion around test execution has been the idea that we want to associate resources with "test threads" to allow limited object re-use. One of the alternatives to this is the pattern of object pooling - create the expensive resources as they're needed but return them to a shared pool and "rent" them for execution.

Microsoft offers an implementation of this pattern via the `Microsoft.Extensions.ObjectPool` package.

Two questions then:
1. Should we consider integrating this package into Reqnroll as a means for supporting our own pooled resources?
2. Should we advise end-users to use this package to pool _their_ resources, such as browser instances?

**Comment by gasparnagy** [comment]
For 2) my answer would be a definite yes, but for 1) I feel that our needs are not complex enough to judge the pain that is introduced by having an additional dependency (which always have problems, because the user code might also depend on that with a different version).

The best would be to update the ReqOverflow sample to use `Microsoft.Extensions.ObjectPool`, so that we can use that as a showcase.
